src/Emulator/Extensions/Mocks/HPSHostController.cs

```diff
@@ -171,6 +174,92 @@ private byte[] GetBytesFromSlave(int count, TimeInterval timeInterval)
             return result;
         }
 
+        // When we need to read for more than 255 bytes, we will need this method
+        private byte[] GetLongBytes(int count, TimeInterval timeInterval) {
+            if (count <= 255) {
+                return GetBytesFromSlave(count, timeInterval);
+            }
+            var result = new byte[count];
+            int rounds = count /255;
+            int remainder = count % 255; 
+            for (int i = 0; i <= rounds; i ++) {
+                int toRead = i == rounds ? remainder : 255;
+                Array.Copy(GetBytesFromSlave(toRead, timeInterval), 0, result, i * 255, toRead);
```

Comment:
What's going to happen if GetBytesFromSlave times out?

Reply:
It returns 0 bytes